PWGHF/D2H/Macros/efficiency.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python3
+
+# Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+# See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+# All rights not expressly granted are reserved.
+
+# This software is distributed under the terms of the GNU General Public
+# License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""
+file: efficiency.py
+brief: script for (acceptance-times-)efficiency computation
+usage: python3 efficiency.py CONFIG
+author: Alexandre Bigot <alexandre.bigot@cern.ch>, Strasbourg University
+"""
+
+import argparse
+import os
+import sys
+from typing import Dict, List, Optional, Union
+
+import numpy as np  # pylint: disable=import-error
+import yaml  # pylint: disable=import-error
+
+# pylint: disable=import-error, no-name-in-module
+from ROOT import (
+    TH1,
+    TH1F,
+    TH2,
+    TCanvas,
+    TEfficiency,
+    TFile,
+    TLatex,
+    TLegend,
+    kAzure,
+    kBlack,
+    kFullCircle,
+    kFullSquare,
+    kOpenSquare,
+    kRed,
+)
+from style_formatter import set_global_style, set_object_style
+
+# options for object style
+COLOR = {"All": kBlack, "Prompt": kRed + 1, "Nonprompt": kAzure + 4}
+MARKERSTYLE = {"All": kFullSquare, "Prompt": kFullCircle, "Nonprompt": kOpenSquare}
+LINEWIDTH = {"All": 1, "Prompt": 1, "Nonprompt": 2}
+
+LINESTYLE = {"All": 1, "Prompt": 2, "Nonprompt": 7}
+
+
+def enforce_list(x: Union[str, List[str]]) -> List[str]:
+    """
+    Helper method to enforce list type
+
+    Parameters
+    ----------
+    - x: a string or a list of string
+
+    Returns
+    ----------
+    - x_list if x was not a list, x itself otherwise
+    """
+
+    if not isinstance(x, list):
+        # handle possible whitespaces in config file entry
+        x_list = x.split(",")
+        for i, element in enumerate(x_list):
+            x_list[i] = element.strip()  # remove possible whitespaces
+        return x_list
+
+    return x
+
+
+def enforce_trailing_slash(path: str) -> str:
+    """
+    Helper method to enforce '/' at the and of directory name
+
+    Parameters
+    ----------
+    - path: some path
+
+    Returns
+    ----------
+    - path with a trailing slash at the end if it was not there yet
+    """
+
+    if path is not None and path[-1] != "/":
+        path += "/"
+
+    return path
+
+
+# pylint: disable=too-many-arguments
+def configure_canvas(
+    name_canvas: str,
+    pt_min: Union[int, float],
+    min_y_axis: Union[int, float],
+    pt_max: Union[int, float],
+    max_y_axis: Union[int, float],
```

Comment:
```suggestion
    pt_min: Union[int, float],
    y_axis_min: Union[int, float],
    pt_max: Union[int, float],
    y_axis_max: Union[int, float],
```
More consistent to have the attribute as a suffix.